FreeRTOS/FreeRTOS-Plus-TCP
# Do not clear pxEndPoint in TCPReturnPacket()

Description
-----------
In response to issue #1077 : 

@XIAO-Q7, thank you for reporting the issue and for the research that you did. Also thanks to @evpopov who encountered the same problem.

I managed to reproduce the problem if I use the following configuration:

- DUT has IP-address 192.168.2.127
- DUT has no gateway (0.0.0.0)
- The TCP remote party has IP address 172.16.0.1

DUT tries to connect to remote device, but `eARPGetCacheEntry()` clears the endpoint, which leads to a dereference of NULL.

Here is a summary of the proposed change:
~~~diff
-     eResult = eARPGetCacheEntry( &ulDestinationIPAddress, &xMACAddress,
-                                  &( pxNetworkBuffer->pxEndPoint ) );
+     eResult = eARPGetCacheEntry( &ulDestinationIPAddress, &xMACAddress,
+                                  &pxEndPoint );

     if( eResult == eARPCacheHit )
     {
         pvCopySource = &xMACAddress;
+        pxNetworkBuffer->pxEndPoint = pxEndPoint;
     }
     else
     {
         pvCopySource = &pxEthernetHeader->xSourceAddress;
     }

+    if( pxNetworkBuffer->pxEndPoint == NULL )
+    {
+        break;
+    }
~~~

I hadn't had the time to check the code coverage yet.

Test Steps
-----------
Use a similar configuration and try to connect to the remote device.

Checklist:
----------




By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
